### PR TITLE
fix: restore Goose sessions by native session id

### PR DESCRIPTION
**Pull request by nikhilachale**
## Summary

Fix Goose session restoration so AO resumes the original interactive Goose conversation using its persisted native session ID.

## Changes

- Restore Goose sessions with:
  `goose session --resume --session-id <agentSessionId>`
- Preserve the configured approval mode through `GOOSE_MODE`.
- Avoid replaying the original task when restoring a native session.
- Avoid passing `--system`, which is not supported by `goose session`.
- Add Goose session-start hook coverage for capturing `agentSessionId`.
- Add adapter tests using a realistic Goose session ID.

## Why

Goose automatically persists interactive sessions in its local SQLite database. AO already captures the native Goose session ID through lifecycle hooks and stores it as `agentSessionId`.

Using that ID allows a terminated AO session to resume the same Goose transcript instead of starting a new conversation or accidentally restoring the most recent unrelated Goose session.

## Restore flow

1. Goose emits a `SessionStart` hook containing its native session ID.
2. AO records that value in the session’s `agentSessionId` metadata.
3. When the AO session is restored, the Goose adapter runs:

   ```bash
   goose session --resume --session-id <agentSessionId>

### CODE REVIEW

File: backend/internal/adapters/agent/goose/goose.go [hunk 1]
```diff
@@ -136,15 +139,7 @@ func (p *Plugin) GetRestoreCommand(ctx context.Context, cfg ports.RestoreConfig)
...
-	cmd = append(gooseModeEnvPrefix(cfg.Permissions), binary, "run")
-	systemPrompt, err := restoreSystemPromptText(cfg)
-	if err != nil {
-		return nil, false, err
-	}
-	if systemPrompt != "" {
-		cmd = append(cmd, "--system", systemPrompt)
-	}
-	cmd = append(cmd, "--resume", "--session-id", agentSessionID)
+	cmd = append(gooseModeEnvPrefix(cfg.Permissions), binary, "session", "--resume", "--session-id", agentSessionID)
```
Comment (illegalcall) [inline backend/internal/adapters/agent/goose/goose.go]: Launch uses `goose run --system <prompt> --interactive`; restore drops `--system`. Since `goose session` cannot take it, please confirm Goose actually **persists** AO's system prompt into the stored session rather than applying it per-invocation — otherwise restored sessions silently lose AO's system instructions. Approval mode is preserved via `GOOSE_MODE`, good.

File: backend/internal/adapters/agent/hooksjson/hooksjson.go [hunk 2]
```diff
@@ -281,15 +280,25 @@ func marshalEvent(rawHooks map[string]json.RawMessage, event string, groups []Ma
 	return nil
 }
 
-func commandExists(groups []MatcherGroup, command string) bool {
+// reconcileHook removes every copy of one AO-owned command before adding the
+// current managed entry under its declared matcher. This lets adapter upgrades
+// change a matcher or timeout without leaving stale or duplicate commands,
+// while preserving every unrelated hook in the affected groups.
+func reconcileHook(groups []MatcherGroup, hook HookEntry, matcher *string) []MatcherGroup {
```
Comment (illegalcall) [inline backend/internal/adapters/agent/hooksjson/hooksjson.go:287]: Solid change and it correctly preserves user hooks, drops emptied groups, and stays idempotent. But `hooksjson` has no test files — this shared infra (touches every adapter) is only covered indirectly via the Qwen migration test. Please add direct unit tests for reconcileHook: matcher-move, dedup across multiple groups, and empty-group cleanup.

**Comment by nikhilachale** [comment]
-Restore again uses goose run --system … --resume --session-id …, ensuring AO instructions are reapplied.
-Added direct reconcileHook tests for matcher movement, deduplication, and empty-group cleanup.

**Review by whoisasx — APPROVED** [review]
Reviewed the Goose/Qwen native session restoration updates and shared hook reconciliation. The implementation preserves unrelated hook entries, migrates AO-managed hook shape safely, and adds focused coverage for the risky paths. No blocking issues found. Looks good to me.

**Review by illegalcall — APPROVED** [review]
